NineChronicles.DataProvider/Store/MySqlStore.cs

```diff
@@ -2297,18 +2297,18 @@ public List<WorldBossRankingModel> GetWorldBossRanking(int raidId, int? queryOff
         {
             using NineChroniclesContext? ctx = _dbContextFactory.CreateDbContext();
             var query = ctx.Set<WorldBossRankingModel>()
-                .FromSqlRaw(@"SELECT `AvatarName`, `HighScore`, `TotalScore`, `Cp`, `Level`, `Address`, `IconId`, row_number() over(ORDER BY `TotalScore` DESC, `UpdatedAt`) as `Ranking` FROM `Raiders` WHERE `RaidId` = {0}", raidId);
+                .FromSqlRaw("SET @prev_score = (SELECT max(`TotalScore`) FROM `Raiders` WHERE `RaidId` = {0}); SET @rank = (SELECT count(*) FROM `Raiders` WHERE `RaidId` = {0} AND `TotalScore` = @prev_score); SELECT `AvatarName`, `HighScore`, `TotalScore`, `Cp`, `Level`, `Address`, `IconId`, @rank := IF(@prev_score = TotalScore, @rank, @rank + 1) as `Ranking` FROM `Raiders` WHERE `RaidId` = {0} ORDER BY `TotalScore` DESC", raidId).ToList();
             if (queryOffset.HasValue)
             {
-                query = query.Skip(queryOffset.Value);
+                query = query.Skip(queryOffset.Value).ToList();
             }
 
             if (queryLimit.HasValue)
             {
-                query = query.Take(queryLimit.Value);
+                query = query.Take(queryLimit.Value).ToList();
```

Comment:
Is it ok if both queryOffset and queryLimit are provided?
In line 2303, query already converted to list.

Reply:
because when ToList was called, IQueryable had to receive the result of the query (full ranked list). I'll add sorting just in case.